src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -279,10 +279,44 @@ private async Task<FunctionMetadataResponse> GetFunctionMetadataAsync(string fun
             {
                 var functionMetadataList = await _functionMetadataProvider.GetFunctionMetadataAsync(functionAppDirectory);
 
-                foreach (var func in functionMetadataList)
+                // depending on the FunctionMetadataProvider used, we may get RpcFunctionMetadata instead of a generated IFunctionMetadata object
+                if (functionMetadataList.First() is RpcFunctionMetadata funcAsRpcMetadata)
                 {
-                    response.FunctionMetadataResults.Add(func);
+                    // simplified payload conversion since we already have the correct type
+                    foreach (var func in functionMetadataList)
```

Comment:
How can we guarantee all of the elements in this list are `RpcFunctionMetadata`?